livre/eq/render/shaders/fragAxis.glsl

```diff
@@ -0,0 +1,55 @@
+#version 420
+
+layout( rgba32f ) uniform image2DRect renderTexture;
+
+uniform ivec4 viewport;
+uniform mat4 invProjectionMatrix;
+uniform vec2 depthRange;
+
+flat in vec3 normal;
+flat in vec3 normal2;
+in vec4 color;
+flat in uint type;
+
+// http://www.opengl.org/wiki/Compute_eye_space_from_window_space.
+vec4 calcPositionInEyeSpaceFromWindowSpace( vec4 windowSpace )
+{
+    vec4 ndcPos;
+    ndcPos.xy = 2.0 * ( windowSpace.xy - viewport.xy - ( viewport.zw / 2.0 )) / viewport.zw;
+    ndcPos.z = 2.0 * ( windowSpace.z - (( depthRange.x + depthRange.y ) / 2.0 )) / ( depthRange.y - depthRange.x );
+    ndcPos.w = 1.0;
+
+    vec4 clipSpacePos = ndcPos / windowSpace.w;
+    vec4 eyeSpacePos = invProjectionMatrix * clipSpacePos;
+    return eyeSpacePos / eyeSpacePos.w;
+}
+
+void main()
+{
+    vec3 pixelEyeSpacePos = vec3( calcPositionInEyeSpaceFromWindowSpace( gl_FragCoord ));
+
+    vec4 dst = imageLoad( renderTexture, ivec2( gl_FragCoord.xy ));
+
+    if( type == 1 )
+    {
+        if(!((( dot( pixelEyeSpacePos, normal ) <= 0.0 ) && ( dot( pixelEyeSpacePos, normal2 ) <= 0.0 )) ||
+             (( dot( pixelEyeSpacePos, normal ) >= 0.0 ) && ( dot( pixelEyeSpacePos, normal2 ) >= 0.0 ))))
```

Comment:
I don't get the "Could you replace both with a single variable a A >= 0 (normalPointsForward, or something similar)"

The code looks like that now:
        if(( dotNormal > 0.0 ) && ( dotNormal2 < 0.0 ) || 
           ( dotNormal < 0.0 ) && ( dotNormal2 > 0.0 ))